internal/fs/sd_windows.go

```diff
@@ -0,0 +1,469 @@
+	var sid *windows.SID
+	err = windows.AllocateAndInitializeSid(&windows.SECURITY_NT_AUTHORITY, 2, windows.SECURITY_BUILTIN_DOMAIN_RID, windows.DOMAIN_ALIAS_RID_ADMINS,
+		0, 0, 0, 0, 0, 0, &sid)
+	if err != nil {
+		return false, errors.Errorf("sid error: %s", err)
+	}
+	token := windows.Token(0)
+	member, err := token.IsMember(sid)
+	if err != nil {
+		return false, errors.Errorf("token membership error: %s", err)
+	}
+	return member, nil
+}
+
+// The code below was adapted from github.com/Microsoft/go-winio under MIT license.
+
+// The MIT License (MIT)
+
+// Copyright (c) 2015 Microsoft
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+var (
+	modadvapi32 = windows.NewLazySystemDLL("advapi32.dll")
+
+	procLookupPrivilegeValueW       = modadvapi32.NewProc("LookupPrivilegeValueW")
+	procAdjustTokenPrivileges       = modadvapi32.NewProc("AdjustTokenPrivileges")
+	procLookupPrivilegeDisplayNameW = modadvapi32.NewProc("LookupPrivilegeDisplayNameW")
+	procLookupPrivilegeNameW        = modadvapi32.NewProc("LookupPrivilegeNameW")
+)
+
+// Do the interface allocations only once for common
+// Errno values.
+const (
+	errnoErrorIOPending = 997
+
+	//revive:disable-next-line:var-naming ALL_CAPS
+	SE_PRIVILEGE_ENABLED = windows.SE_PRIVILEGE_ENABLED
+
+	//revive:disable-next-line:var-naming ALL_CAPS
+	ERROR_NOT_ALL_ASSIGNED syscall.Errno = windows.ERROR_NOT_ALL_ASSIGNED
+)
+
+var (
+	errErrorIOPending error = syscall.Errno(errnoErrorIOPending)
+	errErrorEinval    error = syscall.EINVAL
+
+	privNames     = make(map[string]uint64)
+	privNameMutex sync.Mutex
+)
+
+// PrivilegeError represents an error enabling privileges.
+type PrivilegeError struct {
+	privileges []uint64
+}
+
+// SecurityDescriptorBytesToStruct converts the security descriptor bytes representation
+// into a pointer to windows SECURITY_DESCRIPTOR.
+func SecurityDescriptorBytesToStruct(sd []byte) (*windows.SECURITY_DESCRIPTOR, error) {
```

Comment:
In Microsoft's go-winio package (check https://github.com/microsoft/go-winio/blob/bc421d9108ade8ae2f76c8ef3e90cd4ee690e2c0/backuptar/tar.go#L130), they have deprecated sddl-based usage of SecurityDescriptors and instead use base64 encoded raw descriptors. 

I had updated my implementation from sddl to raw accordingly.
"Maintaining old SDDL-based behavior for backward compatibility. All new tar headers written by this library will have raw binary for the security."

Refer - https://github.com/microsoft/go-winio/commit/0ab2de7e65cca04fcdd6bc8846b0d6e6f2732bd7
"Remove use of SDDL in tar headers
Switching to using straight binary instead of SDDL in tar headers to avoid failures in converting domain SIDs into/out of SDDL."

While testing SDs, they were not getting restored for domain accounts when using SSDL and it worked properly only after updating it to use raw binary as specified in go-winio.

Reply:
> I guess SDs won't work when moving between 32 and 64 bit systems.

According to https://nostarch.com/download/WindowsSecurityInternals_Chapter5.pdf the self-relative SDs, which are used here, are encoded the same on 32/64 bit systems. That also matches that https://learn.microsoft.com/en-us/windows/win32/secauthz/absolute-and-self-relative-security-descriptors says that the format can be stored/transmitted.

So I'm not entirely happy with how we have to serialize the SDs. But as the code is based on https://github.com/microsoft/go-winio/blob/3c9576c9346a1892dee136329e7e15309e82fb4f/sd.go#L118 , the approach is probably fine.